common/src/main/java/org/wildfly/swarm/ts/common/DockerRunner.java

```diff
@@ -0,0 +1,167 @@
+ *
+ */
+public class DockerRunner {
+
+    private static Logger LOGGER = null;
+    private String containerId;
+    private String containerImage;
+    private List<String> ports = new ArrayList<>();
+    private Map<String, String> environmentVariables = new HashMap<>();
+    private String logWait;
+    private long timeoutMillis = 60000;
+
+    public DockerRunner() {
+        LOGGER = Logger.getLogger(DockerRunner.class.getName());
+        LOGGER.setUseParentHandlers(false);
+
+        CustomDockerRunnerFormatter formatter = new CustomDockerRunnerFormatter();
+        ConsoleHandler handler = new ConsoleHandler();
+        handler.setFormatter(formatter);
+
+        LOGGER.addHandler(handler);
+    }
+
+    public DockerRunner(String containerImage, String logWait) {
+        this();
+        this.containerImage = containerImage;
+        this.logWait = logWait;
+    }
+
+    public String getContainerImage() {
+        return containerImage;
+    }
+
+    public void setContainerImage(String containerImage) {
+        this.containerImage = containerImage;
+    }
+
+    public String getLogWait() {
+        return logWait;
+    }
+
+    public void setLogWait(String logWait) {
+        this.logWait = logWait;
+    }
+
+    public void addEnvVar(String key, String value) {
+        environmentVariables.put(key, value);
+    }
+
+    public void addPort(String port) {
+        this.ports.add(port);
+    }
+
+    public List<String> getPorts() {
+        return this.ports;
+    }
+
+    public long getTimeoutMillis() {
+        return timeoutMillis;
+    }
+
+    public void setTimeoutMillis(long timeoutMillis) {
+        this.timeoutMillis = timeoutMillis;
+    }
+
+    public void start() throws Exception {
+
+        containerId = UUID.randomUUID().toString();
+        ProcessBuilder dockerStartProcessBuilder = new ProcessBuilder();
+
+        List<String> parameters = new ArrayList<>();
+
+        parameters.add("docker");
+        parameters.add("run");
+
+        for (String port : ports) {
+            parameters.add("-p");
+            parameters.add(port);
+        }
+
+        for (Entry<String, String> envVar : environmentVariables.entrySet()) {
+            parameters.add("-e");
+            parameters.add("\"" + envVar.getKey() + "=" + envVar.getValue() + "\"");
+        }
+
+        parameters.add("--name");
+        parameters.add(containerId);
+        parameters.add(containerImage);
+
+        dockerStartProcessBuilder.redirectErrorStream(true);
+        dockerStartProcessBuilder.command(parameters);
+
+        Process process = dockerStartProcessBuilder.start();
+
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+
+        executor.execute(() -> {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+                String line = null;
+                while ((line = reader.readLine()) != null) {
+                    LOGGER.info(line);
```

Comment:
Should it probably be just `System.out.println()`?

Reply:
Wanted to do as similar as docker-maven-plugin,  that's the reason printing in custom format

Reply:
The format is good, but not sure about using `java.util.logging` for that.